### PR TITLE
Add unit tests for ChecksCancellation trait

### PR DESCRIPTION
**Pull request by GarethClarridge**
Added `tests/Unit/Traits/ChecksCancellationTest.php` to provide 100% coverage for the `App\Traits\ChecksCancellation` trait. The tests verify happy paths, edge cases (deleted records), and model instance refreshing logic. Verified with PHPUnit and PHPStan.

---
*PR created automatically by Jules for task [3211881956143204152](https://jules.google.com/task/3211881956143204152) started by @GarethClarridge*

### CODE REVIEW

File: tests/Unit/Traits/ChecksCancellationTest.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    public function it_refreshes_the_processing_log_instance(): void
+    {
+        $log = MediaProcessingLog::factory()->create([
+            'status' => ProcessingStatus::Processing,
+        ]);
+
+        $this->subject->processingLog = $log;
+
+        // Update in DB behind the scenes
+        $log->update(['status' => ProcessingStatus::Cancelled]);
```
[the remark this answers is not in the document]
Comment (GarethClarridge) [reply tests/Unit/Traits/ChecksCancellationTest.php:91]: @jules please fix
